### PR TITLE
Fix restoreInitialTopology in SA with multiple SCs

### PR DESCRIPTION
**Pull request by caioluke**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

When restoring the initial topology in SA, we try to close switches/branches in all LfNetworks. When we have more than 1 LfNetwork, the code crashes.

**What is the new behavior (if this is a feature change)?**
We will be able to restore the initial topology in SA with multiple SCs.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by obrix** [comment]
The missing coverage are edge cases very hard to test in a unit test. I feel we can merge as is.

**Review by obrix — APPROVED** [review]

File: src/main/java/com/powsybl/openloadflow/network/impl/Networks.java [hunk 1]
```diff
@@ -222,9 +224,21 @@ public static LfNetworkList load(Network network, LfNetworkParameters networkPar
...
                 for (LfNetwork lfNetwork : lfNetworks) {
+                    // all switches and branches were closed
+                    if (switchAndBranchIdsLeftToClose.isEmpty()) {
+                        break;
+                    }
                     // disable all buses and branches not connected to main component (because of switch to close)
-                    restoreInitialTopology(lfNetwork, modifiedTopoConfig.getSwitchesToClose(), modifiedTopoConfig.getBranchIdsToClose());
+                    restoreInitialTopology(lfNetwork, switchAndBranchIdsLeftToClose);
+                }
+                if (!switchAndBranchIdsLeftToClose.isEmpty()) {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/impl/Networks.java:240]: I don't understand why we could be in that case (with the exception). If we are in that case, it means that we have a bug in the code. Any advice @geofjamg ?

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/network/impl/Networks.java:240]: Either:
  - we are able to add a test case for this case because it is a real case
  - we remove the code because it cannot happen

Comment (caioluke) [reply src/main/java/com/powsybl/openloadflow/network/impl/Networks.java:240]: It is not a real case, I just put it to be sure that we would catch a bug if it happened. I will remove this bit of code.

**Review by annetill — APPROVED** [review]

**Review by geofjamg — APPROVED** [review]
